### PR TITLE
[vtk] Reduce dependencies by adding new features

### LINKED ISSUE #39491: [vtk] Reduce dependencies by adding new features
### Is your feature request related to a problem? Please describe.

I just need vtkImageData. I don't need the 50 dependencies.

### Proposed solution

All dependencies should depend on feature. To keep backward compatibility, new features should be enabled by default.

If you're okay with that, I can work on it.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bansan85**
Fixes #39491.

- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [X] SHA512s are updated for each updated download.
- [X] The "supports" clause reflects platforms that may be fixed by this new version.
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

The CI check all dependencies at once. I want to reduce dependencies of vtk. I will check that the following package can be installed one by one on my computer for triplet x64-windows, x64-windows-static-md, x64-linux, x64-linux-dynamic:

 - [x] itk[core,vtk] : x64-linux, ~~x64-linux-dynamic~~
 - [x] opencascade[core,vtk] : x64-linux
 - [x] opencv[core,vtk] : x64-linux
 - [x] opencv3[core,vtk] : x64-linux
 - [x] opencv4[core,vtk] : x64-linux
 - [x] paraview[core,vtkm] : x64-linux
 - [x] pcl[core,vtk] : x64-linux
 - [x] rtabmap[core,gui] : x64-linux
 - [x] vtk-dicom[core] : x64-linux

The previous tick must be checked before leaving Draft.

### CODE REVIEW

**Comment by bansan85** [comment]
Maybe I was a bit optimist but I think I will be able to remove some dependencies that have a very long compilation time.

**Comment by bansan85** [comment]
itk doesn't build. Update to itk 5.4.0 because build failure is fixed on newer version.

**Comment by bansan85** [comment]
Can't test on x64-linux-dynamic, minc doesn't build.

File: ports/itk/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "itk",
-  "version-string": "5.3rc02",
-  "port-version": 1,
+  "version": "5.4.0",
   "description": "Insight Segmentation and Registration Toolkit (ITK) is used for image processing and analysis.",
   "homepage": "https://github.com/InsightSoftwareConsortium/ITK",
   "license": "Apache-2.0",
```
Comment (Neumann-A) [inline ports/itk/vcpkg.json:6]: This should be a different PR

**Comment by bansan85** [comment]
It takes really long time to build for x64-linux. It "should" be fine for Windows.

File: ports/itk/vcpkg.json (hunk 1 above)
Comment (bansan85) [reply ports/itk/vcpkg.json]: I move it to https://github.com/microsoft/vcpkg/pull/39757

**Comment by bansan85** [comment]
This PR is ready to merge. I don't see how I can lighten it port.

**Comment by JavierMatosD** [comment]
@bansan85, I need a bit more context on what the issue is exactly. Your title claims to be reducing dependencies to improve the build time, but I don't look like the changes reduce the number of dependencies. Can you give a little more context on what you are intending to do? Or perhaps, I am overlooking something?

**Comment by dg0yt** [comment]
IIUC it enables a better effect of top-level `"default-features": false`.

File: ports/seacas/vcpkg.json [hunk 2]
```diff
@@ -38,7 +37,22 @@
     },
     "zlib"
   ],
+  "default-features": [
+    "matio"
+  ],
```
Comment (JavierMatosD) [inline ports/seacas/vcpkg.json:42]: I don't think this should be a default feature. The point of default features is that a user who is transitively using this dependency would consider it "broken" if it did not have this behavior enabled. e.g. a generic file compression library not supporting .zip files.

Comment (JavierMatosD) [reply ports/seacas/vcpkg.json:42]: Additionally, `vcpkg install vtk[all]` should still work.

Comment (Neumann-A) [reply ports/seacas/vcpkg.json:42]: > Additionally, vcpkg install vtk[all] should still work.

is broken since forever due to missing dependencies (https://github.com/microsoft/vcpkg/pull/29260). Mainly needs a decision on tool ports.

Comment (bansan85) [reply ports/seacas/vcpkg.json]: @JavierMatosD I disabled Matio

At first, I enabled it by default to keep compatibility. But I don't think this feature is used by lots of people.

Comment (bansan85) [reply ports/seacas/vcpkg.json]: @JavierMatosD I was waiting that CI finished. I can't read log in progress and I forgot that x64_osx sometimes hangs.
